arch/lkl/kernel/threads.c

```diff
@@ -219,7 +219,7 @@ void threads_cleanup(void)
 		if (t->pid != 1 && !test_ti_thread_flag(ti, TIF_HOST_THREAD))
 			WARN(!(t->flags & PF_KTHREAD),
 			     "non kernel thread task %s\n", t->comm);
-		WARN(t->state == TASK_RUNNING,
+		WARN(t->__state == TASK_RUNNING,
```

Comment:
Not sure if it is needed, but maybe we should use READ_ONCE here.

Reply:
thanks, the part is updated as you suggested.